akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala

```diff
@@ -33,11 +33,24 @@ class SubSource[+Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Source
     new Source(delegate.mergeSubstreams)
 
   /**
+   * Flatten the sub-flows back into the super-flow by performing a merge
+   * with the given breadth limit. This means that only up to `breadth`
+   * substreams will be executed at any given time. Substreams that are not
+   * yet executed are also not materialized, meaning that back-pressure will
+   * be exerted at the operator that creates the substreams when the breadth
+   * limit is reached.
+   */
+  def mergeSubstreamsWithLimit(breadth: Int): Source[Out, Mat] =
```

Comment:
WithDanger

-- 
Cheers,
√
On 12 Dec 2015 18:19, "Roland Kuhn" <notifications@github.com> wrote:

> In akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala
> <https://github.com/akka/akka/pull/19148#discussion_r47434193>:
>
> > @@ -33,11 +33,24 @@ class SubSource[+Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Source
> >      new Source(delegate.mergeSubstreams)
> >
> >    /**
> > +   * Flatten the sub-flows back into the super-flow by performing a merge
> > +   * with the given breadth limit. This means that only up to `breadth`
> > +   * substreams will be executed at any given time. Substreams that are not
> > +   * yet executed are also not materialized, meaning that back-pressure will
> > +   * be exerted at the operator that creates the substreams when the breadth
> > +   * limit is reached.
> > +   */
> > +  def mergeSubstreamsWithLimit(breadth: Int): Source[Out, Mat] =
>
> Since this is deadlock country I prefer to keep this restriction obvious
> and “in your face”.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/19148/files#r47434193>.
>